phoenix-core/src/main/java/org/apache/phoenix/jdbc/PhoenixStatement.java

```diff
@@ -907,10 +909,15 @@ public MutationState execute() throws SQLException {
                     try {
                         FileSystem fs = dynamicJarsDirPath.getFileSystem(conf);
                         List<LiteralParseNode> jarPaths = getJarPaths();
-                        for (LiteralParseNode jarPath : jarPaths) {
-                            File f = new File((String) jarPath.getValue());
-                            fs.copyFromLocalFile(new Path(f.getAbsolutePath()), new Path(
-                                    dynamicJarsDir + f.getName()));
+                        for (LiteralParseNode jarPathNode : jarPaths) {
+                          String jarPathName = (String) jarPathNode.getValue();
+                          File f = new File(jarPathName);
+                          Path dynamicJarsDirPathWithJar = new Path(dynamicJarsDir + f.getName());
+                          // Copy the jar (can be local or on HDFS) to the hbase.dynamic.jars.dir directory.
+                          // Note that this does not support HDFS URIs without scheme and authority.
+                          Path jarPath = new Path(jarPathName);
+                          FileUtil.copy(jarPath.getFileSystem(conf), jarPath, fs, dynamicJarsDirPathWithJar,
```

Comment:
The comment "Note that this does not support HDFS URIs without scheme and authority." I wonder where did you discover that.

If I'm not mistaken FileUtil.copy internally calls fs.copyFromLocalFile without doing file.islocal check.